classes/object_file_system.php

```diff
@@ -472,14 +489,18 @@ public function remove_file($contenthash) {
     }
 
     /**
-     * Copies file by its hash to directory specified at $destinationpath
+     * Copies external file to trashdir by its hash
      *
      * @param string $contenthash file to be copied
-     * @param string $destinationpath destination directory
+     * @return bool
      */
-    public function copy_file_from_hash_to_path($contenthash, $destinationpath) {
-        $path = $this->get_remote_path_from_hash($contenthash);
-        copy($path, $destinationpath);
+    public function copy_external_file_to_trashdir_from_hash($contenthash) {
+        if ($this->deleteexternally) {
+            $currentpath = $this->get_external_path_from_hash($contenthash);
+            $destinationpath = $this->get_external_trash_path_from_hash($contenthash);
+
+            return copy($currentpath, $destinationpath);
```

Comment:
Don't you think it's better to use existing  copy_file_from_hash_to_path method here?

Reply:
copy_file_from_hash_to_path method invokes get_remote_path_from_hash that has a few conditions and slightly different behaviour than copy_external_file_to_trashdir_from_hash.

Reply:
copy_file_from_hash_to_path method invokes get_remote_path_from_hash that has a few conditions and slightly different behaviour than copy_external_file_to_trashdir_from_hash.